### PR TITLE
Deploy to integration

### PR DESCRIPTION
**Pull request by themouette**
Add scripts to create/provision integration machine

* There still is some errors when installing pm2, mostly due to user rights
* ~~Nginx redirection for / is wrong~~
* ~~CSP errors when accessing without specified port~~ <- this was API not accessible

available at http://poc.caliopen.org

### CODE REVIEW

File: integration/bin/setup_machine [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    echo "export PM2_HOME=/home/integ/.pm2" > /etc/profile
+    source /etc/profile
+  fi
+
+  # create a health endpoint on http://localhost:9615
+  su integ -c "pm2 web && pm2 save"
+
+  # Make pm2 a service
+  [[ -e /etc/init.d/pm2-init.sh ]] || PATH=$PATH:/usr/local/bin pm2 startup -u integ
+  service pm2-init start
```
Comment (esion) [inline integration/bin/setup_machine]: is it platform dependant ?

File: integration/bin/setup_machine [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+}
+
+function download_githubfile {
+  wget "${GITHUB_PROJECT_URL}/integration/bin/check-and-restart" -O /root/check-and-restart-caliopen.frontend && \
+    chmod +x /root/check-and-restart-admin
```
Comment (esion) [inline integration/bin/setup_machine]: ` .frontend` seems to be missing

File: integration/bin/check-and-restart [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    npm install bower@latest && \
+    bower install && \
+    npm run release"
+
+echo "Restart $SERVICE" | logger -t CALIOPEN_FRONTEND
+service nginx restart
+service pm2-init restart
+
+rm -f "$lockfile"
+trap - INT TERM EXIT
```
Comment (esion) [inline integration/bin/check-and-restart]: what does it mean ? I  don't speak very well bash

**Comment by esion** [comment]
:+1:

File: integration/bin/setup_machine (hunk 1 above)
Comment (themouette) [inline integration/bin/setup_machine]: up to now it is. Any advice on how to handle this cross platform is welcome (init.d, systemd, systemV...).

File: integration/bin/setup_machine (hunk 2 above)
Comment (themouette) [inline integration/bin/setup_machine]: nice catch

File: integration/bin/check-and-restart (hunk 3 above)
Comment (themouette) [inline integration/bin/check-and-restart]: from my understanding, it removes the signal trap that was set up earlier:

``` sh
trap "rm -f $lockfile; exit" INT TERM EXIT
```

it removes $lockfile on exit

**Pull request merged**